ruler/ruler.go

```diff
@@ -151,12 +151,14 @@ func (r *Ruler) getConfig(userID string) (*config.Config, error) {
 
 // appenderAdapter adapts cortex.SampleAppender to prometheus.SampleAppender
 type appenderAdapter struct {
-	appender cortex.SampleAppender
-	ctx      context.Context
+	distributor *distributor.Distributor
```

Comment:
Not fully - its only implements the Push method.  I thought the coupling was a bit strong here, but there is no other interface just for Push, unless we invent one.